drivers/flash/flash_gecko.c

```diff
@@ -178,6 +184,14 @@ void flash_gecko_page_layout(struct device *dev,
 }
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 
+static const struct flash_parameters *
+flash_gecko_get_parameters(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return (const struct flash_parameters *)&flash_gecko_parameters;
```

Comment:
The cast of the address is unnecessary and confusing.  Please remove from all drivers.

Reply:
Yes. I do not even remember anymore, why I am doing this casting...